### ISSUE TITLE
Union backend reuploading files instead of just changing mod-time

### ISSUE THREAD

**Issue by Dibbyo456** [rclone/rclone#4869]
When last modified time is changed in source Microsoft OneDrive (Union) backend reuploading files instead of just changing the mod-time in destination.

v1.53.3
Ubuntu 20.04
Microsoft OneDrive for Business accounts as Union

```
rclone sync gdrive: union: --onedrive-no-versions -P
```

Log says:

```
2020/12/14 20:22:13 DEBUG : testfile.zip: Modification times differ by 11m12.586s: 2020-12-14 14:10:26.414 +0000 UTC, 2020-12-14 14:21:39 +0000 UTC
```

In this case `testfile.zip` mod-time has changed but the file itself is unchanged. But OneDrive re-uploading the files instead of just changing the mod-time.

**Comment by Dibbyo456** [comment]
Is it because OneDrive using **QuickXorHash** and Google Drive using **MD5**?

**Comment by ncw** [comment]
> Is it because OneDrive using **QuickXorHash** and Google Drive using **MD5**?

Hashes are definitely part of the problem, however I think the problem is that the union backend doesn't support hashes.

@Dibbyo456 what exactly have you got in the union - Google Drive and Onedrive? Can you paste the config for it?

> In this case `testfile.zip` mod-time has changed but the file itself is unchanged. But Union OneDrive re-uploading the files instead of just changing the mod-time.

Rclone will try to prove the files are the same with a checksum and if it can it will update the modification time.

Because the union backend takes different backends which may have different hash support, it doesn't support hashes at all.

Really it should support the intersection of all the backend hash types so if you were joining OneDrive and local it should still support QuickXorHash

This would be relatively easy to add but I need to know what you've got in your union to know if it would help.

**Comment by Dibbyo456** [comment]
> but I need to know what you've got in your union to know if it would help

The config:

```
[gdrive]
type = drive
scope = drive
service_account_file = [redacted]
team_drive = [redacted]

[onedrive1]
type = onedrive
client_id = [redacted]
client_secret = [redacted]
token = [redacted]
drive_type = business

[onedrive2]
type = onedrive
client_id = [redacted]
client_secret = [redacted]
token = [redacted]
drive_type = business

[onedriveunion]
type = union
upstreams = onedrive1: onedrive2:
cache_time = 604800
```

The command:

```
rclone sync gdrive: onedriveunion: --onedrive-no-versions -vv
```

When a file's mod-time is changed in `gdrive`, rclone reuploading the file instead of just updating the mod-time in `onedriveunion`.

**Comment by lostheli** [comment]
How would you like it to work?

Without a reference to determine if the file on gdrive has actually changed, assuming a change in modification time means a change in content seems the safest assumption.  If onedrive and gdrive have different hashes, onedrive cannot be used as the reference (regardless of union), no?  Do you have a special case such as any change results in an increase in file size?  Or file content of existing files is never changed and only new files must be synced?

Otherwise, maybe implement this as a three step process where you download all of the gdrive hashes, compare them to a locally stored set of hashes, and transfer the changed files.  Storing a set of hashes locally solves the reference problem.

**Comment by Dibbyo456** [comment]
yeah, you are right. I wasn't thinking clearly. I think I will have to use `--size-only` flag.
